exporters/cache.go

```diff
@@ -0,0 +1,64 @@
+package exporters
+
+import (
+	"sync"
+	"time"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+var lock = &sync.Mutex{}
+var singleCache CacheBackend
+
+type CacheBackend interface {
+	Set(string, ServiceCacheData)
+	Get(string) (ServiceCacheData, bool)
+	Clear()
+}
+
+// MetricFamily Cache data
+type MFCacheData struct {
+	Time time.Time
+	MF   *dto.MetricFamily
+}
+
+type ServiceCacheData struct {
+	MatricFamilies map[string]MFCacheData
+}
+
+func GetCache() CacheBackend {
+	lock.Lock()
+	defer lock.Unlock()
+	if singleCache == nil {
```

Comment:
question: it seems to me that `singleCache` is private to this module,  so it will be always `nil` (because you cannot override this), even if you import it in `main.go` is that right?

Reply:
yes, but it can be just like

```go
var singleCache CacheBackend = &InMemoryCache{
			CloudCaches: make(map[string]*CloudCache),
		}
.
.
.
func GetCache() CacheBackend {
	lock.Lock()
	defer lock.Unlock()
	return singleCache
}
```

Reply:
okay, yes that's true. But currently it does not have file / redis backend, so it looked weird to me

Reply:
It's a singleton pattern. And there is no benefit we init a private variable in the beginning. Since you shouldn't use it directly. Imagine someone change the map without the lock.

Reply:
okay, what about `func init()` for the module?

Reply:
I found some reading about singleton in Go: http://marcio.io/2015/07/singleton-pattern-in-go/

Reply:
> I found some reading about singleton in Go: http://marcio.io/2015/07/singleton-pattern-in-go/

It's nice to use `sync.Once`. Let me update it.